test/built-ins/Set/prototype/isDisjointFrom/set-like-class-mutation.js

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2023 Anthony Frehner and Kevin Gibbons. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-set.prototype.isdisjointfrom
+description: Set.prototype.isDisjointFrom behavior when a custom Set-like class mutates the receiver
+features: [set-methods]
+includes: [compareArray.js]
+---*/
+
+const baseSet = new Set(["a", "b", "c"]);
+
+const evilSetLike = {
+  size: 3,
+  has(v) {
+    if (v === "a") {
+      baseSet.delete("b");
+      baseSet.delete("c");
+      baseSet.add("b");
```

Comment:
I'm probably overlooking something, but I don't understand what's being tested by deleting and adding `"b"`, as you end up with the same `baseSet` as if you'd only deleted `"c"`. Did you mean for the added element to be `"d"` or something?

Reply:
i assumed it's to test the semantics of visiting after deletion and addition?

Reply:
Me too, but with `"b"` you wouldn't be able to tell the difference if, say, the implementation made a copy of `baseSet`'s initial [[SetData]] and iterated over the correct number of elements. Adding an element that's not `"b"` would make that implementation fail the test.

Reply:
Yeah, from the spec's point of view, deleting and re-adding does not leave you with the same data structure - there's "~empty~" tombstones to mark deleted elements. So the point is just to ensure that the behavior is consistent even if that happens while the method is running.

Reply:
I'm fine with also adding `d` here if you'd like me to though. There's definitely more cases which could be tested, it's just not what I was setting out to test with this.

Reply:
Not a blocker for me, either way. (Some of the other methods' `set-like-class-mutation.js` tests do add elements that weren't in the original [[SetData]], so I think it's unlikely that implementations would be failing this anyway)